xarray/tests/test_dask.py

```diff
@@ -1039,7 +1039,7 @@ def test_map_blocks_error(map_da, map_ds):
     def bad_func(darray):
         return (darray * darray.x + 5 * darray.y)[:1, :1]
 
-    with raises_regex(ValueError, "Length of the.* has changed."):
+    with raises_regex(ValueError, "Received dimension.*"):
```

Comment:
@jhamman this test raises an error for unexpected shapes with no template present.